### DISCUSSION TITLE
Disconnect behaviour

### DISCUSSION THREAD

**Discussion by maxx8888** [Louisvdw/dbus-serialbattery#525] [General]
Hi,

I was wondering what would happen in case there is a Communication fault to the BMS for some Reason.
BMS could be damaged, USB Cable unplugged, Serial Converter dead or stuc, etc...

Shouldn't Multiplus stop Charging/Discharging as quick as possible?
It seems like the last received Value get's stuck and Multiplus just holds that value.

**Comment by mr-manuel** [comment]
The default Victron behaviour would be that it switches to the next available battery monitor or am I wrong?

**Reply by maxx8888** [reply]
Hmm, maybe i've a setting wrong, but Battery Monitor is set to autmatic.
It selects the BMS by default, when unplugging the RS485 Adapter, it just stays at the BMS instead of switching back to Multiplus2.

Anything that i need to consider?

**Reply by mr-manuel** [reply]
Did you try to reboot once? Which Venus OS version do you have?

**Reply by maxx8888** [reply]
2.94, doesn't find any Updates when searching...
Actually I've setup everything on my Bench for testing and rebooting basically all the time :-D

What should happen when disconnecting BMS from Venus OS?
It should immidiatly switch back to Multiplus for Battery Monitor?
Or is a reboot required in between always in order to detect missing BMS?

**Reply by mr-manuel** [reply]
When BMS is connected:

![grafik](https://user-images.githubusercontent.com/12592765/232714264-db70495b-fdc6-46ef-b8cb-0226dde7de53.png)

When I unplug the USB:

![grafik](https://user-images.githubusercontent.com/12592765/232714723-b5f2e4a7-5faa-4a8c-994f-ce4d6b81bcb7.png)

**Reply by maxx8888** [reply]
Strange, that would be perfect, but not happening for me. 
Without reboot in-between?
Are you running on the nightly build?
Maybe I should try that one…

**Reply by mr-manuel** [reply]
I tried Venus OS 2.93 and 3.00~32. Maybe it's 2.94 related. You can boot your old version by goint to `Venus OS GUI -> Settings -> Firmware -> Stored backup firmware -> Selecting the older version`. If this is not chaning anything you can boot again the new version, by following the same steps.

I found notes for 2.94:

DVCC:
* Fix bug where data from the battery could get stuck/freeze until GX devices was rebooted. We've seen this happen in a handful of support cases; and it related to certain timing in the canbus messages. Once properly tested, we might release an interim official version v2.94 that includes this fix soon.
Source: https://community.victronenergy.com/questions/196394/venus-os-v30029-available-for-beta-testing.html

**Reply by mr-manuel** [reply]
I upgraded to 2.94 and it works also there for me. Try the nightly and if that do not work reinstall Venus OS from scratch.

**Reply by maxx8888** [reply]
Perfect!!!!

I've upgraded today to the nigthly and it works.
Detects RS485 disconnects with an Alarm and shuts off after a few seconds.
Detects USB disconnect and shuts off immidiatly.

Thanks :)

**Comment by rperpe** [comment]
I'd prefere to continue discharging in ESS mode when BMS data isn't available anymore since BMS data is expendable in this case from my point of view.

Is this achievable somehow?

**Reply by mr-manuel** [reply]
https://github.com/Louisvdw/dbus-serialbattery/blob/88e5eea93331ae99dc28fcf8b5a40cb866e432e2/etc/dbus-serialbattery/config.default.ini#L21-L27
